fundingmanager_test.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"testing"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+func TestMaxPendingPeerExceed(t *testing.T) {
```

Comment:
In response to the above comments, this test should be changed accordingly. I'd prefer the addition of a test-case using the integration testing framework. 

If you don't think it's too out of scope for this PR, I'd also like to see the `ErrorGeneric` message extended to add a `PendingChannelID` field`. The addition of such a field will let peers communicate errors in the context of a particular pending channel. With this field, once a peer reads an `ErrorGeneric` message with the `PendingChannelID` field set, then they can forward the error to the `fundingManager` who can handle it properly.